src/test/java/org/broadinstitute/hellbender/engine/dataflow/coders/GATKReadCoderUnitTest.java

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.hellbender.engine.dataflow.coders;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import org.broadinstitute.hellbender.utils.dataflow.DataflowUtils;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+
+public class GATKReadCoderUnitTest extends BaseTest {
+
+    private GATKRead makeGoogleRead( final int uuid, final String name, final int start, final int length ) {
+        return ArtificialReadUtils.createGoogleBackedReadWithUUID(new UUID(0, uuid), name, start, length);
+    }
+
+    public GATKRead makeSamRead( final int uuid, final String name, final int start, final int length ) {
+        return ArtificialReadUtils.createSamBackedReadWithUUID(new UUID(0, uuid), name, start, length);
+    }
+
+    @DataProvider(name = "reads")
+    public Object[][] makeReads() {
+        final List<GATKRead> googleReads = Arrays.asList(makeGoogleRead(1, "google1", 1, 10), makeGoogleRead(2, "google2", 20, 40));
+        final List<GATKRead> samReads = Arrays.asList(makeSamRead(1, "sam1", 1, 10), makeSamRead(2, "sam2", 20, 40));
+
+        return new Object[][] {
+                { googleReads },
+                { samReads }
```

Comment:
Added one, but now I get an error: `com.google.cloud.dataflow.sdk.coders.CannotProvideCoderException: Cannot provide coder for Create: The elements are not all of the same class.`